### PR TITLE
fix(manage): dont error when restoring deleted achievement credit

### PR DESCRIPTION
**Pull request by wescopeland**
Resolves https://discord.com/channels/310192285306454017/310195377993416714/1314263872303005750.

To reproduce the bug:
* Add any form of achievement credit for a user.
* Delete the credit.
* Try to re-add the same credit for the user.

This surfaces a SQL error because it's a soft-deleted record that actually needs to be restored.

### CODE REVIEW

File: app/Filament/Resources/AchievementResource/RelationManagers/AuthorshipCreditsRelationManager.php [hunk 1]
```diff
@@ -58,6 +58,24 @@ public function table(Table $table): Table
...
+                            ->whereAchievementId($achievement->id)
+                            ->whereUserId($userId)
+                            ->whereTask($task)
+                            ->first();
+
+                        if ($existingRecord) {
+                            if ($existingRecord->trashed()) {
+                                $existingRecord->restore();
+
+                                return $existingRecord;
```
Comment (Jamiras) [inline app/Filament/Resources/AchievementResource/RelationManagers/AuthorshipCreditsRelationManager.php:74]: this return is redundant with the return on line 77.

**Review by Jamiras — COMMENTED** [review]
Noticed while testing this: Creating a new credit with a specific date creates a record with today's date. It is possible to then edit the record to select the specific date.

File: app/Filament/Resources/AchievementResource/RelationManagers/AuthorshipCreditsRelationManager.php (hunk 1 above)
Comment (wescopeland) [reply app/Filament/Resources/AchievementResource/RelationManagers/AuthorshipCreditsRelationManager.php]: Removed in latest.

**Comment by wescopeland** [comment]
> Noticed while testing this: Creating a new credit with a specific date creates a record with today's date.

Good catch. The input date should now be respected in latest.

**Pull request merged**